src/core/noinit.jl

```diff
@@ -0,0 +1,37 @@
+# Some algorithms don't support creating a cache and doing `solve!`, this unfortunately
+# makes it difficult to write generic code that supports caching. For the algorithms that
+# don't have a `__init` function defined, we create a "Fake Cache", which just calls
+# `__solve` from `solve!`
+@concrete mutable struct NonlinearSolveNoInitCache{iip, timeit} <:
+                         AbstractNonlinearSolveCache{iip, timeit}
+    prob
+    alg
+    args
+    kwargs::Any
```

Comment:
does this effect type inference? I don't see why this wouldn't be concrete as well.